src/util/arith_tools.cpp

```diff
@@ -281,10 +283,31 @@ void mp_max(mp_integer &a, const mp_integer &b)
 /// \param bit_index: index (0 is the least significant)
 bool get_bitvector_bit(const irep_idt &src, std::size_t bit_index)
 {
-  // The representation is binary, using '0'/'1',
-  // most significant bit first.
-  PRECONDITION(bit_index < src.size());
-  return src[src.size() - 1 - bit_index] == '1';
+  // The representation is hex, most significant nibble first.
+  const auto nibble_index = bit_index >> 2;
+
+  if(nibble_index >= src.size())
+    return false;
```

Comment:
Isn't this changing the behaviour and suppressing some error cases?

Reply:
Yes, we do loose the ability to check whether 'bit-index' is within bounds. Retaining that would require passing the width to get_bitvector_bit.

Reply:
So maybe it should be passed in? It seems like a reasonable part of the contract?

Reply:
Ok, will pass that in.

Reply:
#3196

Reply:
This needs to be changed now that #3196 has been merged to re-instate the `PRECONDITION`.